nes-systests/systest/src/SystestExecutor.cpp

```diff
@@ -62,7 +62,15 @@ using namespace std::literals;
 
 namespace NES
 {
-SystestConfiguration readConfiguration(int argc, const char** argv)
+SystestExecutor::SystestExecutor(SystestConfiguration config) : config(std::move(config))
+{
+}
+
+SystestExecutor::SystestExecutor(int argc, const char** argv) : SystestExecutor(parseConfiguration(argc, argv))
+{
+}
+
+SystestConfiguration SystestExecutor::parseConfiguration(int argc, const char** argv)
```

Comment:
IMHO, parsing configuration in a type called `SystestExecutor` cries for decomposition. An executor should execute, not parse.